### PR TITLE
Cancel search while typing

### PR DESCRIPTION
**Pull request by PVince81**
- [x] REQUIRES https://github.com/nextcloud/spreed/pull/4706 because of future conflicts

The code is in the very last commit if you want to have a peek.

### CODE REVIEW

**Comment by nickvergessen** [comment]
Sounds like a good idea

File: src/services/conversationsService.js [hunk 1]
```diff
@@ -91,29 +91,24 @@ const checkTalkVersionHash = function(response) {
 /**
  * Fetch listed conversations
  * @param {string} searchTerm The string that will be used in the search query.
+ * @param {object} options options
  */
-const searchListedConversations = async function(searchTerm) {
-	const response = await axios.get(generateOcsUrl('apps/spreed/api/v3', 2) + 'listed-room', {
+const searchListedConversations = async function(searchTerm, options) {
+	return axios.get(generateOcsUrl('apps/spreed/api/v3', 2) + 'listed-room', Object.assign(options, {
```
Comment (PVince81) [inline src/services/conversationsService.js:97]: note: await is gone because redundant (if you leave it here eslint will warn about it)

Comment (nickvergessen) [reply src/services/conversationsService.js:97]: 🍿

**Review by nickvergessen — APPROVED** [review]

**Comment by PVince81** [comment]
I was so focused on testing the cancellability that I forgot to test if the search actually works. A copy-paste issue had sneaked in, fixed here: https://github.com/nextcloud/spreed/pull/4835

**Pull request merged**